sycl/test/on-device/basic_tests/partition_supported.cpp

```diff
@@ -0,0 +1,92 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+/* Check that:
+1) [info::device::partition_properties]: returns the partition properties
+supported by this SYCL device; a vector of info::partition_property. If this
+SYCL device cannot be partitioned into at least two sub devices then the
+returned vector **must be empty**.
+
+2) [create_sub_devices()]: If the SYCL device
+does not support info::partition_property::partition_by_affinity_domain or the
+SYCL device does not support the info::partition_affinity_domain provided, an
+exception with the **feature_not_supported error code must be thrown**.
+*/
+
+#include <CL/sycl.hpp>
+
+/** returns true if the device supports a particular affinity domain
+ */
+static bool
+supports_affinity_domain(const cl::sycl::device &dev,
+                         cl::sycl::info::partition_property partitionProp,
+                         cl::sycl::info::partition_affinity_domain domain) {
+  if (partitionProp !=
+      cl::sycl::info::partition_property::partition_by_affinity_domain) {
+    return true;
+  }
+  auto supported =
+      dev.get_info<cl::sycl::info::device::partition_affinity_domains>();
+  for (cl::sycl::info::partition_affinity_domain dom : supported) {
+    if (dom == domain) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/** returns true if the device supports a particular partition property
+ */
+static bool
+supports_partition_property(const cl::sycl::device &dev,
+                            cl::sycl::info::partition_property partitionProp) {
+  auto supported = dev.get_info<cl::sycl::info::device::partition_properties>();
+  for (cl::sycl::info::partition_property prop : supported) {
+    if (prop == partitionProp) {
+      return true;
+    }
+  }
+  return false;
+}
+
+int main() {
```

Comment:
I suspect this test passes even without the change in the L0 plugin.
If so, I suggest writing two tests: one which makes sure that correct PI calls are done(if it doesn't exist already) and second which checks that results of PI API calls are aligned with direct calls to L0 API( see https://github.com/intel/llvm/blob/sycl/sycl/unittests/pi/cuda/test_queue.cpp#L145 )

Reply:
> I suspect this test passes even without the change in the L0 plugin.

No, it does not pass without the plugin fix. Indeed, Level-Zero plugin returned that the partitioning was always supported, so the subsequent call create_sub_devices() was returning "0" devices, and the check for "<2" fired.

Reply:
Ok.